cluster-operator/src/test/java/io/strimzi/operator/cluster/operator/assembly/KafkaAssemblyOperatorWithPoolsKRaftMockTest.java

```diff
@@ -280,6 +280,89 @@ public void testReconcileWithControllerRelevantConfigChange(VertxTestContext con
+                    assertThat(spsControllers, is(notNullValue()));
+
+                    spsControllers.getSpec().getPods().stream().map(PodSetUtils::mapToPod).forEach(pod -> {
+                        loggingConfigurationAnnos.put(pod.getMetadata().getName(), pod.getMetadata().getAnnotations().get(Annotations.ANNO_STRIMZI_LOGGING_APPENDERS_HASH));
+                    });
+
+                    StrimziPodSet spsBrokers = supplier.strimziPodSetOperator.client().inNamespace(namespace).withName(CLUSTER_NAME + "-brokers").get();
+                    assertThat(spsBrokers, is(notNullValue()));
+
+                    spsBrokers.getSpec().getPods().stream().map(PodSetUtils::mapToPod).forEach(pod -> {
+                        loggingConfigurationAnnos.put(pod.getMetadata().getName(), pod.getMetadata().getAnnotations().get(Annotations.ANNO_STRIMZI_LOGGING_APPENDERS_HASH));
+                    });
+
+                    // Update Kafka with dynamically changeable option that is not controller relevant => controller pod annotations should not change
+                    Crds.kafkaOperation(client).inNamespace(namespace).withName(CLUSTER_NAME)
+                            .edit(k -> new KafkaBuilder(k).editSpec().editKafka().withNewInlineLogging().withLoggers(Map.of("kafka.root.logger.level", "DEBUG")).endInlineLogging().endKafka().endSpec().build());
+                })))
+                .compose(v -> operator.reconcile(new Reconciliation("test-trigger", Kafka.RESOURCE_KIND, namespace, CLUSTER_NAME)))
+                .onComplete(context.succeeding(v -> context.verify(() -> {
+                    StrimziPodSet spsControllers = supplier.strimziPodSetOperator.client().inNamespace(namespace).withName(CLUSTER_NAME + "-controllers").get();
+                    assertThat(spsControllers, is(notNullValue()));
+
+                    spsControllers.getSpec().getPods().stream().map(PodSetUtils::mapToPod).forEach(pod -> {
+                        // Controller annotations should differ
+                        assertThat(pod.getMetadata().getAnnotations().get(Annotations.ANNO_STRIMZI_LOGGING_APPENDERS_HASH), is(not(loggingConfigurationAnnos.get(pod.getMetadata().getName()))));
+                    });
+
+                    StrimziPodSet spsBrokers = supplier.strimziPodSetOperator.client().inNamespace(namespace).withName(CLUSTER_NAME + "-brokers").get();
+                    assertThat(spsBrokers, is(notNullValue()));
+
+                    spsBrokers.getSpec().getPods().stream().map(PodSetUtils::mapToPod).forEach(pod -> {
+                        // Broker annotations should be the same
+                        assertThat(pod.getMetadata().getAnnotations().get(Annotations.ANNO_STRIMZI_LOGGING_APPENDERS_HASH), is(loggingConfigurationAnnos.get(pod.getMetadata().getName())));
+                    });
+
+                    // Update Kafka with dynamically changeable controller relevant option => controller pod annotations should change
```

Comment:
Unless I'm misunderstanding, I think this should non dynamic 
```suggestion
                    // Update Kafka with not dynamically changeable option => both controller and broker pod annotations should change
```